src/h5_async_vol.c

```diff
@@ -9053,13 +9961,20 @@ async_dataset_write(async_instance_t *aid, H5VL_async_t *parent_obj, hid_t mem_t
     bool                        is_blocking = false;
     hbool_t                     acquired    = false;
     unsigned int                mutex_count = 1;
+    herr_t                      return_val;
 
     func_enter(__func__, NULL);
 
     assert(aid);
     assert(parent_obj);
     assert(parent_obj->magic == ASYNC_MAGIC);
+    
+
+    return_val =
+        async_dataset_write_merge(aid, parent_obj, mem_type_id, mem_space_id, file_space_id, plist_id, buf);
 
+    if (return_val == 1)
+        goto done; 
```

Comment:
add #ifdef ENABLE_WRITE_MERGE ... #endif before 9973 and after 9977, so that the merge only happens when enabled at compile time